src/main/java/org/commcare/util/EncryptionKeyServiceProvider.java

```diff
@@ -0,0 +1,44 @@
+package org.commcare.util;
+
+import java.util.NoSuchElementException;
+import java.util.ServiceLoader;
+
+/**
+ * Utility class responsible for finding implementations of IEncryptionKeyProvider during runtime
+ * and loading them in memory
+ *
+ * @author avazirna
+ */
+
+public class EncryptionKeyServiceProvider {
+    private static EncryptionKeyServiceProvider serviceProvider;
+    private ServiceLoader<IEncryptionKeyProvider> loader;
+
+    private EncryptionKeyServiceProvider() {
+        loader = ServiceLoader.load(IEncryptionKeyProvider.class);
+    }
+
+    public static EncryptionKeyServiceProvider getInstance() {
+        if (serviceProvider == null) {
+            serviceProvider = new EncryptionKeyServiceProvider();
+        }
+        return serviceProvider;
+    }
+
+    public IEncryptionKeyProvider serviceImpl(boolean useKeyStoreIfAvailable) {
```

Comment:
@shubham1g5 I added a flag to determine the preference in terms of using or not the platform KeyStore, however, this is not doesn't prevent a non preferred implementation from being returned when there is no KeyStore available.